src/Discord/Parts/Channel/Message.php

```diff
@@ -79,6 +80,125 @@ public function reply($text)
     {
         return $this->channel->sendMessage("{$this->author}, {$text}");
     }
+    
+    /**
+     * React to a message.
+     *
+     * @param Emoji $emoticon  The emoticon to react with. (example: '👎', custom: ':michael:251127796439449631')
+     *
+     * @return \React\Promise\Promise
+     */
+    public function react($emoticon)
+    {
+        $deferred = new Deferred();
+
+        $this->http->put(
+            "channels/{$this->channel->id}/messages/{$this->id}/reactions/{$emoticon}/@me"
+        )->then(
+            function ($response) use ($deferred) {
```

Comment:
Changing this to match the rest of the library with `\React\Partial\bind_right($this->resolve, $deferred)`.